indico/modules/events/forms.py

```diff
@@ -135,3 +136,43 @@ class UnlistedEventsForm(IndicoForm):
     authorized_creators = PrincipalListField(_('Authorized users'), [HiddenUnless('enabled', preserve_data=True)],
                                              allow_external_users=True, allow_groups=True,
                                              description=_('These users may create unlisted events.'))
+
+
+class DataRetentionSettingsForm(IndicoForm):
+    minimum_data_retention = TimeDeltaField(_('Minimum data retention period'), units=('weeks',),
+                                            description=_('Specify the minimum data retention period (in weeks) '
+                                                          'configurable for registrations. This includes the '
+                                                          'retention period of individual registration form fields.'),
+                                            render_kw={'placeholder': _('Indefinite'), 'min': 1})
```

Comment:
This placeholder is not correct, leaving it empty means you get the default of `1` which means the field should probably be required since setting it to `1` is the same as the current behavior (that if you set a retention period, it cannot be less than 1 week)